core/src/consensus/fork_choice.rs

```diff
@@ -54,6 +64,35 @@ pub trait ForkChoice {
     ) -> Vec<Self::ForkChoiceKey>;
 }
 
+fn last_vote_able_to_land(
```

Comment:
Pulled this out of the select_forks_failed_switch_threshold function, flipped the polarity, un-nested things